### PR TITLE
Add Bonsai C4 as a platform board file

### PR DESCRIPTION
**Pull request by infinityis**
This creates Bonsai C4 as a board file (as recommended [here](https://discord.com/channels/440868230475677696/440868230475677698/1013346652247035944)), which makes the ProMicro to Bonsai C4 converter work without having to configure hardware, change/add code, etc.

This also removes the Bonsai C4 template code from the keyboards/customMK folder which demonstrated how to configure the hardware, since that is no longer needed, and updates the converter documentation accordingly.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Conflicts will need to be resolved, otherwise it will fall into the next develop cycle.

**Comment by infinityis** [comment]
Conflicts have been resolved

File: platforms/chibios/boards/BONSAI_C4/configs/config.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#    define EXTERNAL_EEPROM_SPI_CLOCK_DIVISOR 8 // 96MHz / 8 = 12MHz; max supported by MB85R64 is 20MHz
+#    define EXTERNAL_EEPROM_BYTE_COUNT 8192
+#    define EXTERNAL_EEPROM_PAGE_SIZE 64 // does not matter for FRAM, just sets the RAM buffer size in STM32F chip
+#    define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 8191
+#endif
+
+// External flash configuration
+#ifndef EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN
+#    define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN PAL_LINE(GPIOB, 12)
+#    define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 1  // 96MHz; max supported by W25Q128JV is 133MHz
```
Comment (tzarc) [inline platforms/chibios/boards/BONSAI_C4/configs/config.h:37]: Minimum clock divisor on STM32 is `/2` -- it'd work with this code in place, as it'll default to `/2` anyway, but for clarity it should probably be fixed up.
```suggestion
#    define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 2  // 48MHz; max supported by W25Q128JV is 133MHz
```

**Review by tzarc — CHANGES_REQUESTED** [review]
Just a small fix, rest LGTM.

**Comment by infinityis** [comment]
Fixed, thank you!

**Pull request merged**